### ISSUE TITLE
blob

### ISSUE THREAD

**Issue by pandaro** [pandaro/Shard#1]
@Tarendai @eronoobos @Beherith
so this is one of my average testing
you can see many of my idea especially inside economic things, tell me what do you think about:
-splitting tasqueue in subfile
-use function composite for economic [things(](https://github.com/pandaro/Shard/blob/blob/data/ai/BAR/taskqueues.lua#L470) (break much loops)
-a possible new bestfactory function [bestfactory](https://github.com/pandaro/Shard/blob/blob/data/ai/BAR/taskqueuebehaviour.lua#L865) i'm working on move it to new behaviour file
-counthandle and countbehaviour now reach many info
-reclaimbehaviour have a new thing called "clean":([here](https://github.com/pandaro/Shard/blob/blob/data/ai/BAR/reclaimbehaviour.lua#L77)) basically some builders remove old economic stuff when useless(like solar when a fusion is builded)
-unitlist now have some other info

yes i know this is a useless and confused blob. I put inside it many thing, if some of you have time to spend to tell me what a good idea and what is not...

**Comment by eronoobos** [comment]
When I wrote the BA config, I bloated taskqueues and taskqueuebehaviour into a monster. There are many things we can do to make it a more efficient, but ultimately I think it needs to be changed to a centralized system that hands out build plans to builders. Rather than checking every entry in a queue, the centralized system would determine from current conditions what needs to be built. The problem is, it would be a lot of work to refactor everything.

If you're considering writing a new BestFactory function because the current one often does not give the best factory: BestFactory is only as good as the MapHandler's analysis of the map, which often is not great.

**Comment by pandaro** [comment]
@eronoobos 

> If you're considering writing a new BestFactory function because the current one often does not give the best factory: BestFactory is only as good as the MapHandler's analysis of the map, which often is not great.

yes i considering a new BestFactory function, but not only, as you can see [here](https://springrts.com/phpbb/viewtopic.php?f=15&t=22582&start=768), i wil write 

> - A separate behaviour that takes over from the task behaviour when it's deemed that a factory should be built, using whatever logic or system is desired

to do this i start from mapHandler and try to do something of more accuracy.

as an aside this will be a slimming for taskqueuebehaviour.

A question about the idea of a centralized brain for think to build:
How can you can update this system if currently you have a updater system bounded with builders?
